### PR TITLE
Fix wire

### PR DESCRIPTION
**Pull request by dlyr**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Some kind of bug fix, and small improvements.

* **What is the current behavior?** (You can also link to an open issue here)
When wireframe rendering is triggered, a snapshot of current geometry is made, and note updated if geometry is updated

* **What is the new behavior (if this is a feature change)?**
Animated/updated geometry trigger wireframe update with observer mechanism

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
nope

* **Other information**:
will be updated with #710 which ease update and consistency.
Also this PR add two minior fix commits

### CODE REVIEW

File: src/Engine/Rendering/ForwardRenderer.cpp [hunk 1]
```diff
@@ -238,8 +320,8 @@ void ForwardRenderer::renderInternal( const Data::ViewingParameters& renderData
 
     GL_ASSERT( glDrawBuffers( 1, buffers ) ); // Draw color texture
 
-    // Radium V2 : this render loop might be greatly improved by inverting light and objects loop
-    // Make shaders bounded only once, minimize full stats-changes, ...
+    // Radium V2 : this render loop might be greatly improved by inverting light and objects
+    // loop Make shaders bounded only once, minimize full stats-changes, ...
```
Comment (MathiasPaulin) [inline src/Engine/Rendering/ForwardRenderer.cpp:324]:
```suggestion
    // loop. 
    // Make shaders bounded only once, minimize full stats-changes, ...
```

File: src/Engine/Rendering/ForwardRenderer.cpp [hunk 2]
```diff
@@ -220,9 +302,9 @@ void ForwardRenderer::renderInternal( const Data::ViewingParameters& renderData
     {
         ro->render( zprepassParams, renderData, DefaultRenderingPasses::Z_PREPASS );
     }
-    // Transparent objects are rendered in the Z-prepass, but only their fully opaque fragments (if
-    // any) might influence the z-buffer Rendering transparent objects assuming that they discard
-    // all their non-opaque fragments
+    // Transparent objects are rendered in the Z-prepass, but only their fully opaque fragments
+    // (if any) might influence the z-buffer Rendering transparent objects assuming that they
```
Comment (MathiasPaulin) [inline src/Engine/Rendering/ForwardRenderer.cpp:306]:
```suggestion
    // (if any) might influence the z-buffer. 
    // Rendering transparent objects assuming that they
```

File: src/Engine/Rendering/ForwardRenderer.cpp [hunk 3]
```diff
@@ -322,8 +404,8 @@ void ForwardRenderer::renderInternal( const Data::ViewingParameters& renderData
         GL_ASSERT( glEnable( GL_DEPTH_TEST ) );
     }
     // Volumetric pass
-    // Z-test is enabled but z-write must be disable to allow access to the z-buffer in the shader.
-    // This pass render in its own FBO and copy the result to the main colortexture
+    // Z-test is enabled but z-write must be disable to allow access to the z-buffer in the
+    // shader. This pass render in its own FBO and copy the result to the main colortexture
```
Comment (MathiasPaulin) [inline src/Engine/Rendering/ForwardRenderer.cpp:408]:
```suggestion
    // shader. 
    // This pass render in its own FBO and copy the result to the main colortexture
```

File: src/Engine/Rendering/ForwardRenderer.cpp [hunk 4]
```diff
@@ -371,106 +453,50 @@ void ForwardRenderer::renderInternal( const Data::ViewingParameters& renderData
...
-                                int i1 = index[i];
-                                int i2 = index[( i + 1 ) % s];
-                                if ( i1 > i2 ) std::swap( i1, i2 );
-                                indices.emplace_back( i1, i2 );
-                            }
-                        }
-                    }
+                    if ( tp->getRenderMode() ==
+                         Data::AttribArrayDisplayable::MeshRenderMode::RM_TRIANGLES )
+                    { setupLineMesh( disp, tp->getCoreGeometry() ); }
```
Comment (MathiasPaulin) [inline src/Engine/Rendering/ForwardRenderer.cpp:483]: There is here unneeded code duplication (from line 473 to 483), could be replaced by 
```.cpp
                auto processLineMesh = []( auto m, std::shared_ptr<Data::LineMesh>& disp ) {
                    if ( m->getRenderMode() ==
                         Data::AttribArrayDisplayable::MeshRenderMode::RM_TRIANGLES )
                    { setupLineMesh( disp, m->getCoreGeometry() ); }
                };
                if ( tm ) { processLineMesh( tm, disp ); }
                if ( tp ) { processLineMesh( tp, disp );
```

**Review by MathiasPaulin — COMMENTED** [review]
once rebased on Radium master and suggestions integrated, will be OK for me.

File: src/Engine/Rendering/ForwardRenderer.cpp (hunk 4 above)
Comment (dlyr) [reply src/Engine/Rendering/ForwardRenderer.cpp]: done

**Review by MathiasPaulin — APPROVED** [review]

**Pull request merged**